futures-util/src/future/remote_handle.rs

```diff
@@ -36,17 +44,87 @@ impl<T> RemoteHandle<T> {
     pub fn forget(self) {
         self.keep_running.store(true, Ordering::SeqCst);
     }
+
+    /// Poll whether result is available, but do not extract result.
+    pub fn poll_complete(&mut self, cx: &mut Context<'_>) -> Poll<()> {
+        self.rx.poll_complete(cx)
+    }
+
+    /// Create a future only polling for completion without returning the result
+    ///
+    /// Use [`recv`] to extract final result after completion is ready.
+    ///
+    /// [`recv`]: struct.RemoteHandle.html#method.recv
+    pub fn completion_handle(&mut self) -> RemoteCompletionHandle<'_> {
+        RemoteCompletionHandle {
+            remote_handle: self,
+        }
+    }
+
+    /// Try to extract final result
+    ///
+    /// Returns `None` if underlying future is still running.
+    ///
+    /// # Panics
+    ///
+    /// Panics in the same way as polling `RemoteHandle`.
+    pub fn try_recv(&mut self) -> Option<T> {
+        Some(unwrap_result(self.rx.try_recv().transpose()?))
+    }
+
+    /// Extract final result
+    ///
+    /// # Panics
+    ///
+    /// Panics if underlying future is still running in addition to the
+    /// ways it would panic when polling `RemoteHandle`.
+    pub fn recv(mut self) -> T {
```

Comment:
Is there some name that might better indicate this panics if still running, `unwrap_recv` maybe? (I feel like there's a good chance users might assume this does a blocking receive).

Reply:
Hm. Or I could simply remove the method - `try_recv` should be good enough after all.